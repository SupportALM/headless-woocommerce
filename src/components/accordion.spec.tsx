import { render } from '@testing-library/react';

import type { AccordionItem } from '@components/accordion';
import { Accordion } from '@components/accordion';

const accordionItems: AccordionItem[] = [
  {
    title: 'Test Content 1',
    content: '<div>Test Content</div>',
    isOpen: true,
  },
  {
    title: 'Test Content 2',
    content: '<div>Test Content 2</div>',
  },
];

describe('<Accordion />', () => {
  it('captures snapshot', () => {
    const { container } = render(<Accordion data={accordionItems} />);
    expect(container).toMatchSnapshot();
  });
});
