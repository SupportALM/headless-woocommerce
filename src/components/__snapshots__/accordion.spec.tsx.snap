// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> captures snapshot 1`] = `
<div>
  <div>
    <div
      class="border-t w-full"
      data-headlessui-state="open"
    >
      <h3>
        <button
          aria-controls="headlessui-disclosure-panel-:r1:"
          aria-expanded="true"
          class="border-b border-b-border-brand-second-gray group relative w-full py-5 flex justify-between items-center text-left"
          data-headlessui-state="open"
          id="Test Content 1"
          type="button"
        >
          <span
            class="text-black font-medium"
          >
            Test Content 1
          </span>
          <span
            class="ml-6 flex items-center mr-1"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="18 15 12 9 6 15"
              />
            </svg>
          </span>
        </button>
      </h3>
      <div
        class="w-full"
        data-headlessui-state="open"
        id="headlessui-disclosure-panel-:r1:"
      >
        <div
          class="tab-content w-full border-b border-brand-second-gray py-6 tab-test content 1"
        >
          &lt;div&gt;Test Content&lt;/div&gt;
        </div>
      </div>
    </div>
    <div
      class="border-t w-full"
      data-headlessui-state=""
    >
      <h3>
        <button
          aria-expanded="false"
          class="border-b border-b-border-brand-second-gray group relative w-full py-5 flex justify-between items-center text-left"
          data-headlessui-state=""
          id="Test Content 2"
          type="button"
        >
          <span
            class="text-black font-medium"
          >
            Test Content 2
          </span>
          <span
            class="ml-6 flex items-center mr-1"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </span>
        </button>
      </h3>
    </div>
  </div>
</div>
`;
